src/orion/service/client/base.py

```diff
@@ -0,0 +1,146 @@
+"""Minimal API to run workon"""
+import logging
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+import requests
+from bson import ObjectId
+
+from orion.core.io.database import DuplicateKeyError
+from orion.core.utils.exceptions import BrokenExperiment, CompletedExperiment
+from orion.core.utils.flatten import unflatten
+from orion.core.worker.trial import AlreadyReleased
+
+log = logging.getLogger(__name__)
+
+
+class ExperiementIsNotSetup(Exception):
+    """Raised when the API is missing an experiment name"""
+
+
+class RemoteException(Exception):
+    """Raised from a request that failed"""
+
+
+@dataclass
+class RemoteExperiment:
+    """Simple read only experiment"""
+
+    euid: str
+    name: str
+    space: dict
+    version: str
+    mode: str
+    working_dir: str
+    metadata: dict
+    max_trials: int
+
+
+@dataclass
+class RemoteTrial:
```

Comment:
Maybe this should be turned into a standard dataclass for trials exposed to users. The current version misses some attributes such as the objectives, constraints and stats.